### PR TITLE
MULE-18169: Race condition generating NullPointerException during OAuth token refresh

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: src/test/java/org/mule/test/oauth/internal/DancerConfigTestCase.java [hunk 1]
```diff
@@ -448,6 +454,82 @@ public void multipleDancersShareTokensStoreMigrateContext()
...
+    final OAuthAuthorizationCodeDancerBuilder builder =
+        service.authorizationCodeGrantTypeDancerBuilder(lockFactory, tokensStore, el);
+    builder.clientCredentials("clientId", "clientSecret");
+    minimalAuthCodeConfig(builder);
+    final AuthorizationCodeOAuthDancer dancer = startDancer(builder);
+
+    final ResourceOwnerOAuthContext contextOwner = new DefaultResourceOwnerOAuthContext(new ReentrantLock(), "owner");
+    tokensStore.put("owner", contextOwner);
+
+    int removerThreadsCount = 150, getterThreadsCount = 170;
```
Comment (elrodro83) [inline src/test/java/org/mule/test/oauth/internal/DancerConfigTestCase.java]: 2 lines please

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Pull request merged**